relayer/src/chain/cosmos.rs

```diff
@@ -864,15 +869,19 @@ impl CosmosSdkChain {
                             ))];
 
                             // Otherwise, try to resolve transaction hash to the corresponding events.
-                        } else if let Ok(events_per_tx) =
-                            self.query_txs(QueryTxRequest::Transaction(QueryTxHash(response.hash)))
+                        } else if let Ok((events_per_tx, cosmos_events_per_tx)) =
+                            self.query_extended_events_for_tx(QueryTxHash(response.hash))
```

Comment:
>     1. Does this mean that _all_ Hermes CLIs will do `query_extended_events_for_tx` as of this PR?

Yes. I chose this as a way to avoid copying `wait_for_block_commits` to do almost exactly the same job, while not exposing these internal changes in the generic `send_messages_and_wait_commit` API.

>     2. Does this have any performance implications or does it affect the return value for any other Hermes CLI command (aside from `tx raw upgrade chain)?

I expect the performance implications to be negligible: Cosmos and IBC events are [parsed out of the same list](https://github.com/informalsystems/ibc-rs/blob/fd3d09e717df8f7d480b11baba25fe73d9974318/relayer/src/chain/cosmos.rs#L965) of ABCI events, which is still retrieved in a single query (in an intermediate change it was not so, but I have reworked it to avoid this exact performance hit).

As for the return value, the only way it would change the output of any other CLI command or other relayer logic using `send_messages_and_wait_commit`, is if the caller expected an IBC event in response to the message, but got no IBC events and some recognized Cosmos event(s) instead. In this case, the return value would be an empty list (as the Cosmos events get discarded in the generic API). I expect there are no such cases at the moment, and if any appear, that would indicate something wrong in our expectations about the ABCI chain.